containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -108,6 +108,25 @@ def get_account_id() -> str:
     """
     return boto3.client("sts").get_caller_identity()["Account"]
 
+def search_string_for_regex(string: str, regex: str) -> str:
+    """Search a string for a regex pattern and return the first result"""
+    database_name_search = re.search(regex, string)
+    if database_name_search:
+        return database_name_search.groups()[0]
+    else:
+        raise ValueError(f"{regex} not found in {string}")
+
+
+def database_name_regex() -> str:
+    return """database_name=([^\/]*)\/"""  # noqa: W605
+
+
+def table_name_regex() -> str:
+    return """table_name=([^\/]*)\/"""  # noqa: W605
+
+
+def extraction_timestamp_regex() -> str:
+    return """(extraction_timestamp=[^\/]*)\/"""  # noqa: W605
```

Comment:
There is no need to surpress W605 here, this warning is due to python trying to intepret `\/` as an escape sequence before it gets to the regex library.

For regexes you should use raw strings like `r"(extraction_timestamp=[^\/]*)\/"`

Reply:
This particular one already exists as a precompiled regex at the start of the module, 

```python
EXTRACTION_TIMESTAMP_REGEX = re.compile(
    r"^(.*)/(extraction_timestamp=)([0-9TZ]{1,16})/(.*)$"
)
```

I think this is doing much the same thing except it's anchored at the end of the string - so we should choose one regex or the other.

It's unlikely to make a significant performance difference at the moment, but in general precompiling the regex with re.compile will speed things up if the regex gets called a lot of times (e.g. in deeply nested loops).